torch/utils/cpp_extension.py

```diff
@@ -157,6 +157,18 @@ def _join_rocm_home(*paths) -> str:
 The detected CUDA version ({0}) mismatches the version that was used to compile
 PyTorch ({1}). Please make sure to use the same CUDA versions.
 '''
+CUDA_MISMATCH_WARN = '''
+
+                               !! WARNING !!
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+The detected CUDA version ({0}) mismatches the version that was used to compile
+PyTorch ({1}). Please make sure to use the same CUDA versions, otherwise some
+errors may appear during compilation/runtime.
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+                              !! WARNING !!
+'''
```

Comment:
```suggestion
CUDA_MISMATCH_WARN = "The detected CUDA version ({0}) mismatches the version that was used to compile
PyTorch ({1}). Most likely this shouldn't be a problem"
```

As discussed in the original feedback thread this is not user-friendly - printing the huge full screen WARNING is super-scary for users and they will be seeking help.

Every time there is any problem with CUDA extension users tend to quote these WARNINGS, since they look like errors. 

If we agreed that most of the time minor version mismatch should be not a problem, let's make it so. Hence the  proposed low-profile warning.

Reply:
except I don't know if it was used elsewhere, so perhaps this one should be called explicitly `CUDA_MISMATCH_MINOR_VERSION_WARN`